telemetry-core/src/main/java/com/newrelic/telemetry/metrics/MetricBuffer.java

```diff
@@ -103,6 +118,106 @@ public MetricBatch createBatch() {
     return new MetricBatch(metrics, this.commonAttributes);
   }
 
+  /**
+   * Creates an ArrayList of MetricBatch objects from the contents of this buffer, then clears the
+   * contents of this buffer.
+   *
+   * <p>{@link Metric Metrics} are added to a MetricBatch. When each metric is added, the size (in
+   * bytes) of the metric is calculated. When the total size of the metrics in the batch exceeds the
+   * MAX_UNCOMPRESSED_BATCH_SIZE, the current batch is sent to New Relic, and a new MetricBatch is
+   * created. This process repeats until all metrics are removed from the queue.
+   *
+   * @return An ArrayList of MetricBatch objects. Each {@link MetricBatch} in the ArrayList contains
+   *     an immutable collection of {@link Metric Metrics}.
+   */
+  public ArrayList<MetricBatch> createBatches() {
+    logger.debug("Creating metric batch.");
+    ArrayList<MetricBatch> metricBatches = new ArrayList<>();
+    Collection<Metric> metricsInBatch = new ArrayList<>();
+
+    int currentUncompressedBatchSize = 0;
+
+    // Construct JSON for common attributes and add to uncompressed batch size
+
+    String commonJson = generateCommonJSON();
+
+    currentUncompressedBatchSize += commonJson.getBytes(StandardCharsets.UTF_8).length;
+
+    // JSON generation + calculating payload size
+
+    Metric curMetric;
+    while ((curMetric = this.metrics.poll()) != null) {
```

Comment:
Ok, it looks like this flows this way and I think there is a problem:

get curMetric  (if this metric gets added to the batch, it will exceed size) 
-- uncompressed size is not over, skip to next if block
-- curMetric is a Count, calculate size and add it to the current uncompressized Size. The current uncompressed Size now exceeds max...but it will be too late by the time this gets checked again?
-- curMetric is added to Batch (The batch now exceeds the size) 
-- Uncompressed is greater than Max...create Batch(which is too big) and add to list.

Did I follow this incorrectly?